ffi/diplomat/src/timezone.rs

```diff
@@ -33,143 +22,123 @@ pub mod ffi {
-        pub fn try_from_offset_seconds(seconds: i32) -> DiplomatResult<ICU4XGmtOffset, ICU4XError> {
-            GmtOffset::try_from_offset_seconds(seconds)
-                .map(Into::into)
-                .map_err(Into::into)
-                .into()
+        pub fn try_set_gmt_offset_seconds(
+            &mut self,
+            offset_seconds: i32,
+        ) -> DiplomatResult<(), ICU4XError> {
+            match GmtOffset::try_from_offset_seconds(offset_seconds) {
+                Ok(v) => {
+                    self.0.gmt_offset = Some(v);
+                    Ok(())
+                }
+                Err(e) => Err(ICU4XError::from(e)),
+            }
+            .into()
         }
 
-        #[diplomat::rust_link(icu::timezone::GmtOffset::offset_seconds, FnInStruct)]
-        pub fn offset_seconds(&self) -> i32 {
-            GmtOffset::from(self).offset_seconds()
+        pub fn gmt_offset_seconds(&self) -> DiplomatResult<i32, ()> {
+            self.0
+                .gmt_offset
+                .map(|v| v.offset_seconds())
+                .ok_or(())
+                .into()
         }
 
         #[diplomat::rust_link(icu::timezone::GmtOffset::is_positive, FnInStruct)]
-        pub fn is_positive(&self) -> bool {
-            GmtOffset::from(self).is_positive()
+        pub fn is_gmt_offset_positive(&self) -> DiplomatResult<bool, ()> {
+            self.0.gmt_offset.map(|v| v.is_positive()).ok_or(()).into()
         }
 
         #[diplomat::rust_link(icu::timezone::GmtOffset::is_zero, FnInStruct)]
-        pub fn is_zero(&self) -> bool {
-            GmtOffset::from(self).is_zero()
+        pub fn is_gmt_offset_zero(&self) -> DiplomatResult<bool, ()> {
+            self.0.gmt_offset.map(|v| v.is_zero()).ok_or(()).into()
         }
 
         #[diplomat::rust_link(icu::timezone::GmtOffset::has_minutes, FnInStruct)]
-        pub fn has_minutes(&self) -> bool {
-            GmtOffset::from(self).has_minutes()
+        pub fn gmt_offset_has_minutes(&self) -> DiplomatResult<bool, ()> {
+            self.0.gmt_offset.map(|v| v.has_minutes()).ok_or(()).into()
         }
 
         #[diplomat::rust_link(icu::timezone::GmtOffset::has_seconds, FnInStruct)]
-        pub fn has_seconds(&self) -> bool {
-            GmtOffset::from(self).has_seconds()
+        pub fn gmt_offset_has_seconds(&self) -> DiplomatResult<bool, ()> {
+            self.0.gmt_offset.map(|v| v.has_seconds()).ok_or(()).into()
         }
-    }
 
-    impl ICU4XOptionalGmtOffset {
-        pub fn create_some(value: ICU4XGmtOffset) -> Self {
-            Self {
-                _private_data: value._private_data,
-            }
+        pub fn try_set_time_zone_id(&mut self, id: &str) -> DiplomatResult<(), ICU4XError> {
+            todo!()
         }
 
-        pub fn create_none() -> Self {
-            Self {
-                _private_data: i32::MAX,
-            }
+        pub fn time_zone_id<'a>(&'a self) -> DiplomatResult<&'a str, ()> {
```

Comment:
issue: we should return writeable here for safety

Reply:
[60f5ffb](https://github.com/unicode-org/icu4x/pull/2442/commits/60f5ffbaedc0c95044be07fcea1872884f8a8dec)